packages/common-scripts/src/anyone_can_pay.ts

```diff
@@ -32,11 +32,13 @@ import {
   isAcpScript,
   SECP_SIGNATURE_PLACEHOLDER,
 } from "./helper";
+import { CellCollectorConstructor, CellCollectorType } from "./type";
 const { ScriptValue } = values;
 const { CKBHasher, ckbHash, readBigUInt128LECompatible } = utils;
 
-export class CellCollector implements CellCollectorType {
-  private cellCollector: CellCollectorType;
+export const CellCollector: CellCollectorConstructor = class CellCollector
+  implements CellCollectorType {
+  private cellCollector: BaseCellCollectorType;
   private config: Config;
```

Comment:
maybe we can implement more than one interface

```ts
export class CellCollector implements CellCollectorType, CellCollectorConstructor {}
```

Reply:
we can not  implements `CellCollectorConstructor` cause it's [static side](https://www.typescriptlang.org/docs/handbook/interfaces.html#difference-between-the-static-and-instance-sides-of-classes).  and also we need `CellCollector`  from '`anyone_one_can_pay`' ,'`secp256k1_blake160_multisig`' and '`secp256k1_blake160`' as same type. so we write `CellCollector: CellCollectorConstructor` here